components/HTTP/src/Websocket.cpp

```diff
@@ -47,423 +46,560 @@ static esp_err_t _json_parse_hex(cJSON* root, std::uint8_t *buf,
         }
 
-        break;
-    case WsMsgType::Message:
-        ESP_LOGI(TAG, "!!!!!msg");
+        /* Copy byte string */
+        cbor_value_get_uint64(&int_it, out);
+        return ESP_OK;
+    }
 
-        err = messageRecipient(ws_pkt.payload, &client_uid);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "Error getting message uids %s", esp_err_to_name(err));
-            return;
-        };
 
-        err = sendWsData(client_uid, ws_pkt, uart);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "Error sending message %s", esp_err_to_name(err));
-            return;
-        };
+    esp_err_t parseMessage(std::uint8_t * buffer, chat_msg_t * msg, size_t length)
+    {
+        CborParser parser;
+        CborValue it;
 
-        break;
-    case WsMsgType::Status:
-        ESP_LOGI(TAG, "!!!!!status");
+        if (cbor_parser_init(buffer, length, 0, &parser, &it) != CborNoError) {
+            ESP_LOGE(TAG, "chat: couldn't parse chat cbor input\n");
+            return ESP_FAIL;
+        }
 
-        err = messageRecipient(ws_pkt.payload, &client_uid);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "Error getting message uids %s", esp_err_to_name(err));
-            return;
-        };
+        if (!cbor_value_is_map(&it)) {
+            ESP_LOGE(TAG, "chat: not a map\n");
+            return ESP_FAIL;
+        }
 
-        err = sendWsData(client_uid, ws_pkt, uart);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "Error sending message %s", esp_err_to_name(err));
-            return;
-        };
+        /* Parse fromUID */
+        if (_parse_chat_id(&it, "fromUID", &msg->from_uid) < 0) {
+            ESP_LOGE(TAG, "chat: fromUID is invalid!\n");
+            return ESP_FAIL;
+        }
 
-        break;
-    case WsMsgType::Action:
-        ESP_LOGI(TAG, "!!!!!action");
+        /* Parse toUID */
+        if (_parse_chat_id(&it, "toUID", &msg->to_uid) < 0) {
+            ESP_LOGE(TAG, "chat: toUID is invalid!\n");
+            return ESP_FAIL;
+        }
 
-        err = messageRecipient(ws_pkt.payload, &client_uid);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "Error getting message uids %s", esp_err_to_name(err));
-            return;
-        };
+        /* Parse msgID */
+        if (_parse_chat_id(&it, "msgID", &msg->msg_id) < 0) {
+            ESP_LOGE(TAG, "chat: msgID is invalid!\n");
+            return ESP_FAIL;
+        }
 
-        err = sendWsData(client_uid, ws_pkt, uart);
-        if (err != ESP_OK) {
-            ESP_LOGE(TAG, "Sending message: %s", esp_err_to_name(err));
-            return;
-        };
+        /* Parse message content */
+        if (_parse_msg_content(&it, "msg", &msg->msg) < 0) {
+            ESP_LOGE(TAG, "chat: msg is invalid!\n");
+            return ESP_FAIL;
+        }
+
+        /* Parse timestamp */
+        if (_parse_uint64(&it, "timestamp", &msg->timestamp) < 0) {
+            ESP_LOGE(TAG, "chat: invalid timestamp!\n");
```

Comment:
check if you can use some error code available in: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/error-codes.html